src/main/java/com/spotify/reaper/service/SegmentRunner.java

```diff
@@ -327,15 +327,15 @@ boolean canRepair(RepairSegment segment, String keyspace, JmxProxy coordinator,
                 + " Run id '{}'", hostName, segment.getRunId());
       JmxProxy hostProxy = null;
       try{
-        Optional<HostMetrics> hostMetrics;
+        Optional<HostMetrics> hostMetrics = Optional.absent();
         try{
           hostProxy = context.jmxConnectionFactory.connect(hostName);
           connected = true;
+          hostMetrics = getMetricsForHost(hostName, hostProxy);
```

Comment:
hostProxy can be `null` (either `context.jmxConnectionFactory.connect(hostName)` throws an exception or returns null.

before getMetricsForHost..)` would throw a NPE that wasn't caught.
now it's caught, and subsequently `gotMetricsForAllHosts` is correctly set to false.